app/crud/categories.php

```diff
@@ -1,245 +1,277 @@
-    <div class="modal fade" id="editmodal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel"
-        aria-hidden="true">
-        <div class="modal-dialog" role="document">
-            <div class="modal-content">
-                <div class="modal-header">
-                    <h5 class="modal-title" id="exampleModalLabel">Edit Data</h5>
-                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
-                        <span aria-hidden="true">&times;</span>
-                    </button>
-                </div>
+            <form action="categories_operation.php" method="POST">
+                <div class="modal-body">
+                    <?php
+                        require_once '../models/Competition.php';
 
-                <form action="categories_operation.php" method="POST">
-                    <div class="modal-body">
-                        <input type="hidden" name="update_id" id="update_id">
-                        <div class="form-group">
-                            <label>Competition_ID</label>
-                            <input type="number" min="1" max="2" name="competition_id" id="competition_id" class="form-control" placeholder="Select your Competition_ID 1(sports) or 2(litmusda)" autocomplete="off" required>
-                        </div>
-
-                        <div class="form-group">
-                            <label>Slug</label>
-                            <input type="text" name="slug" id="slug" class="form-control" placeholder="Enter your Slug">
-                        </div>
-
-                        <div class="form-group">
-                            <label>Title</label>
-                            <input type="text" name="title" id="title" class="form-control" placeholder="Enter your Title">
-                        </div>
+                        $page_url = $_SERVER['REQUEST_URI'];
+                        $entity_category = '';
+                        $competitions = Competition::all();
+                        foreach ($competitions as $competition) {
+                            $competition_url = strtolower(str_replace(' ', '-', $competition->getSlug()));
+                            if (strpos($page_url, $competition_url) !== false) {
+                                $entity_category = $competition->getTitle();
+                                break;
+                            }
+                        }
+                    ?>
+                    <div class="form-group">
+                        <label>Competition</label>
+                        <select name="competition_id" class="form-control" required <?php if (!empty($entity_category)) echo 'disabled'; ?>>
+                            <option value="">Select Competition</option>
+                            <?php foreach ($competitions as $competition) {
+                                $selected = ($competition->getTitle() == $entity_category) ? 'selected' : '';
```

Comment:
Then, check if a competition `slug` is equal to the `$entity_category` to make an `<option>` element selected.

```suggestion
                                $selected = ($competition->getSlug() == $entity_category) ? 'selected' : '';
```